socorro/processor/rules/mozilla.py

```diff
@@ -953,38 +955,32 @@ def predicate(self, raw_crash, dumps, processed_crash, proc_meta):
         # Beta and aurora versions send the wrong version in the crash report for
         # certain products
         product = processed_crash.get("product", "")
-        release_channel = processed_crash.get("release_channel", "")
-        return (
-            product.lower() in self.SUPPORTED_PRODUCTS
-            and release_channel.lower()
-            in (
-                "beta",
-                "aurora",
-            )
+        release_channel = processed_crash.get("release_channel", "").lower()
+        return product in self.SUPPORTED_PRODUCTS and release_channel in (
+            "beta",
+            "aurora",
         )
 
     def action(self, raw_crash, dumps, processed_crash, processor_meta):
-        product = processed_crash.get("product", "").strip().lower()
+        product = processed_crash["product"]
         build_id = processed_crash.get("build", "").strip()
-        release_channel = processed_crash.get("release_channel").strip()
+        release_channel = processed_crash["release_channel"].lower()
 
         # Only run if we've got all the things we need
-        if (
-            product
-            and build_id
-            and release_channel
-            and product in self.SUPPORTED_PRODUCTS
```

Comment:
We don't need this check because the predicate wouldn't pass if this weren't true and then this would never run.